src/Func.h

```diff
@@ -114,14 +115,15 @@ class Func : public Obj
 		return Invoke(&zargs);
 		}
 
-	// Add a new event handler to an existing function (event).
+	// Various ways to add a new event handler to an existing function
+	// (event).
+	void AddBody(const detail::FunctionIngredients& ingr, detail::StmtPtr new_body = nullptr);
```

Comment:
Here's what I've drafted as a replacement comment, let me know what you think:

```
        // Various ways to add a new event handler to an existing function
        // (event).  The usual version to use is the first with its default
        // parameter.  All of the others are for use by script optimization,
        // as is a non-default second parameter to the first method, which  
        // overrides the function body in "ingr".
```

Regarding non-detail API ... sigh.  As I framed above, I'd like to understand if there's a clean way of introducing publicly available `detail` methods alongside non-`detail` ones. Sometimes it's not feasible to instead use `friend` constructs because of how scattered the calls are across different functions.

Reply:
Thanks for the sketches, they've helped my appreciate of the various issues. I'll add the above comment with the next push.